packages/opentelemetry-context-asynchooks/src/asynchooks-context.ts

```diff
@@ -0,0 +1,126 @@
+/**
```

Comment:
This file is pretty much the same as Opencensus: https://github.com/census-instrumentation/opencensus-node/blob/master/packages/opencensus-core/src/internal/cls-ah.ts, i didn't saw the point of re-implementing differently since this one is already used in production with Census.